gpytorch/models/exact_prediction_strategies.py

```diff
@@ -73,6 +76,78 @@ def _exact_predictive_covar_inv_quad_form_root(self, precomputed_cache, test_tra
         # where S S^T = (K_XX + sigma^2 I)^-1
         return test_train_covar.matmul(precomputed_cache)
 
+    def get_fantasy_strategy(self, inputs, targets, full_inputs, full_targets, full_output):
+        full_mean, full_covar = full_output.mean, full_output.lazy_covariance_matrix
+
+        batch_shape = full_inputs[0].shape[:-2]
+
+        full_mean = full_mean.view(*batch_shape, -1)
+        num_train = self.num_train
+
+        # Evaluate fant x train and fant x fant covariance matrices, leave train x train unevaluated.
+        fant_fant_covar = full_covar[..., num_train:, num_train:]
+        mvn = self.likelihood(MultivariateNormal(torch.zeros(1), fant_fant_covar), inputs)
```

Comment:
You'll want to use a `.type_as(full_mean)` on the zeros here to make it work on the GPU / with double dtype.  Also, why is this `torch.zeros(1)` and not something like `torch.zeros(*fant_fant_covar.shape[:-2], fant_fant_covar.shape[-1])` ? Or is this all broadcasted appropriately?

Reply:
The mean doesn't much matter but there's  better choice of dummy tensor to use:
```suggestion
        mvn = self.likelihood(MultivariateNormal(fant_mean, fant_fant_covar), inputs)
```